subprojects/docs/src/docs/userguide/javaPlugin.adoc

```diff
@@ -539,6 +539,55 @@ The incremental compilation at a high level:
 * Because of type erasure, the incremental compiler is not able to recognize when a type is only used in a type parameter, and never actually used in the code. For example, imagine that you have the following code: `List&lt;? extends A&gt; list = Lists.newArrayList();` but that no member of `A` is in practice used in the code, then changes to `A` will not trigger recompilation of the class. In practice, this should very rarely be an issue.
 
 
+=== Incremental annotation processing
+
+Starting with Gradle 4.7, the incremental compiler also supports incremental annotation processing.
+Annotation processors need to opt-in to this feature, otherwise they will result in full recompilation.
+As a user you can see which annotation processors are causing full recompilations in the `--info` log.
+
+==== Making an annotation processor incremental
+
+Gradle supports incremental compilation for two common categories of annotation processors.
+Processors that don't fit these categories will result in full recompilation as before.
```

Comment:
> Processors that don't fit these categories

Is there a specific category or example we can name here? 

Also: "as before" is superfluous here.

Reply:
Well, any processors that do any of the things listed as limitations below. E.g. those that have untracked side effects, those that are nondeterministic (yes, that's a thing), those that read/write resources. I'll try to fit this in here.